ray-operator/controllers/ray/utils/serve_httpclient.go

```diff
@@ -0,0 +1,160 @@
+	r.client = http.Client{}
+	r.dashboardURL = "http://" + url
+}
+
+func (r *RayDashboardClient) GetDeployments() (string, error) {
+	req, err := http.NewRequest("GET", r.dashboardURL+DeployPath, nil)
+	if err != nil {
+		return "", err
+	}
+
+	resp, err := r.client.Do(req)
+
+	if err != nil {
+		return "", err
+	}
+	defer resp.Body.Close()
+
+	body, _ := ioutil.ReadAll(resp.Body)
+
+	return string(body), nil
+}
+
+func (r *RayDashboardClient) UpdateDeployments(specs []rayv1alpha1.ServeConfigSpec) error {
+
+	servingClusterDeployments := ServingClusterDeployments{
+		Deployments: r.convertServeConfig(specs),
+	}
+
+	deploymentJson, err := json.Marshal(servingClusterDeployments)
+
+	if err != nil {
+		return err
+	}
+
+	req, err := http.NewRequest("PUT", r.dashboardURL+DeployPath, bytes.NewBuffer(deploymentJson))
+	if err != nil {
+		return err
+	}
+	req.Header.Set("Content-Type", "application/json")
+
+	resp, err := r.client.Do(req)
+
+	if err != nil {
+		return err
+	}
+	defer resp.Body.Close()
+
+	return nil
+}
+
+func (r *RayDashboardClient) GetDeploymentsStatus() (*rayv1alpha1.ServeDeploymentStatuses, error) {
+	req, err := http.NewRequest("GET", r.dashboardURL+StatusPath, nil)
+	if err != nil {
+		return nil, err
+	}
+
+	resp, err := r.client.Do(req)
+
+	if err != nil {
+		return nil, err
+	}
+	defer resp.Body.Close()
+
+	body, _ := ioutil.ReadAll(resp.Body)
+
+	var serveStatuses rayv1alpha1.ServeDeploymentStatuses
+	if err = json.Unmarshal(body, &serveStatuses); err != nil {
+		return nil, err
+	}
+
+	return &serveStatuses, nil
+}
+
+func (r *RayDashboardClient) convertServeConfig(specs []rayv1alpha1.ServeConfigSpec) []ServeConfigSpec {
+	serveConfigToSend := make([]ServeConfigSpec, len(specs))
+
+	for i, config := range specs {
+		serveConfigToSend[i] = ServeConfigSpec{
+			Name:                      config.Name,
+			ImportPath:                config.ImportPath,
+			InitArgs:                  config.InitArgs,
+			InitKwargs:                config.InitKwargs,
+			NumReplicas:               config.NumReplicas,
+			RoutePrefix:               config.RoutePrefix,
+			MaxConcurrentQueries:      config.MaxConcurrentQueries,
+			UserConfig:                config.UserConfig,
+			AutoscalingConfig:         config.AutoscalingConfig,
+			GracefulShutdownWaitLoopS: config.GracefulShutdownWaitLoopS,
+			GracefulShutdownTimeoutS:  config.GracefulShutdownTimeoutS,
+			HealthCheckPeriodS:        config.HealthCheckPeriodS,
+			HealthCheckTimeoutS:       config.GracefulShutdownTimeoutS,
+			RayActorOptions: RayActorOptionSpec{
+				RuntimeEnv:        config.RayActorOptions.RuntimeEnv,
+				NumCpus:           config.RayActorOptions.NumCpus,
+				NumGpus:           config.RayActorOptions.NumGpus,
+				Memory:            config.RayActorOptions.Memory,
+				ObjectStoreMemory: config.RayActorOptions.ObjectStoreMemory,
+				Resources:         config.RayActorOptions.Resources,
+				AcceleratorType:   config.RayActorOptions.AcceleratorType,
```

Comment:
Might not be be completely trivial to get the AcceleratorType argument working correctly (the arg was designed for AWS VMs), but we'll get there...

Reply:
Right now, all the fields are synced up the Ray repo. I will keep it updating once api endpoint finalized.

Reply:
Yeah, sounds good. Just noting that this one might not work initially.